tests/std/tests/GH_004597_self_swap/test.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <utility>
+
+struct swap_counter {
+    unsigned int* pcnt_ = nullptr;
+
+    friend _CONSTEXPR20 void swap(swap_counter& lhs, swap_counter& rhs) noexcept {
+        std::swap(lhs.pcnt_, rhs.pcnt_);
+        if (lhs.pcnt_ != nullptr) {
+            ++(*lhs.pcnt_);
+        }
+        if (rhs.pcnt_ != nullptr) {

```

Comment:
We conventionally insert a newline between "non-chained" `if`-statements, which helps to make control flow clearer.